imfilebrowser.h

```diff
@@ -90,8 +90,24 @@ namespace ImGui
         // default value is 0 (the first type filter)
         void SetCurrentTypeFilterIndex(int index);
 
+        // when ImGuiFileBrowserFlags_EnterNewFilename is set
+        // this function will pre-fill the input dialog with a filename.
+        void SetInputName(std::basic_string<char> input) noexcept
+        {
+            if((flags_ & ImGuiFileBrowserFlags_EnterNewFilename)) {
+                if (input.size() >= static_cast<size_t>(INPUT_NAME_BUF_SIZE)) {
+                    // if input doesn't fit trip off characters
+                    input.resize(INPUT_NAME_BUF_SIZE - 1);
+            }
+            std::copy(input.begin(), input.end(), inputNameBuf_->begin());
+            inputNameBuf_->at(input.size()) =
+                '\0'; // I think we need to add the null.
```

Comment:
curious about why this simple expression occupies a whole new line

Reply:
removed comment and put the '\0' on the same line as above.